core/src/main/java/org/apache/calcite/sql/SqlCaseStmt.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql;
+
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.util.ImmutableNullableList;
+
+import java.util.List;
+import java.util.Objects;
+/**
+ * Parse tree for a {@code SqlCaseStmt}.
+ */
+public class SqlCaseStmt extends SqlCall {
+  public static final SqlSpecialOperator OPERATOR =
+      new SqlSpecialOperator("CASE", SqlKind.CONDITIONAL_STATEMENT);
```

Comment:
Should we use a more specific SqlKind for CONDITIONAL_STATEMENT (same for the existing SqlIfStmt type)?

Reply:
Added in https://github.com/googleinterns/calcite/pull/263/commits/59ec4c1cd314b7c1b1161ac20c8f9361d1c3c2ec